### ISSUE TITLE
[Bug Report] El-Cascader-Panel Options Not Updating After Call to Await

### ISSUE THREAD

**Issue by MarkoH17** [element-plus/element-plus#2837]
### Element Plus version
1.0.2-beta.69

### OS/Browsers version
macOS 11.5 / Google Chrome v92.0

### Vue version
3.2.0-beta.7

### Reproduction Link
https://codepen.io/markoh2017/pen/WNjgdZg?editors=1000

### Steps to reproduce
Options for "el-cascader-panel" do not reflect new values after they are modified by an async method.

### What is Expected?
The cascader panel should show new values as they are added to the options array.

### What is actually happening?
Data pushed to the options array by an async method are not shown in the cascader panel.

**Comment by MarkoH17** [comment]
See the below screenshot for an example:
![image](https://user-images.githubusercontent.com/8473031/128242383-19d48764-36c4-4aa6-949f-d42183bc4c40.png)

**Comment by SimonaliaChen** [comment]
The cause here is not the async method. If you run the same code in `mounted` hook, you will find out all `addOption` methods not work. The real cause is you mutate `options` with `push` method rather than replacing which supposed to be totally fine. But I add a equality judgement to prevent duplicated render in https://github.com/element-plus/element-plus/pull/2344.

**Comment by JeremyWuuuuu** [comment]
> The cause here is not the async method. If you run the same code in `mounted` hook, you will find out all `addOption` methods not work. The real cause is you mutate `options` with `push` method rather than replacing it with a new array which supposed to be totally fine. But I add an equality judgement to prevent duplicated render in #2344. You can simply use a new array to change the options for now. I'll fix it ASAP

Weird, should we watch the options array to make it more natural?

**Comment by SimonaliaChen** [comment]
> > The cause here is not the async method. If you run the same code in `mounted` hook, you will find out all `addOption` methods not work. The real cause is you mutate `options` with `push` method rather than replacing it with a new array which supposed to be totally fine. But I add an equality judgement to prevent duplicated render in #2344. You can simply use a new array to change the options for now. I'll fix it ASAP
> 
> Weird, should we watch the options array to make it more natural?

According to Vue documentation:

**Comment by JeremyWuuuuu** [comment]
> > > The cause here is not the async method. If you run the same code in `mounted` hook, you will find out all `addOption` methods not work. The real cause is you mutate `options` with `push` method rather than replacing it with a new array which supposed to be totally fine. But I add an equality judgement to prevent duplicated render in #2344. You can simply use a new array to change the options for now. I'll fix it ASAP
> > 
> > 
> > Weird, should we watch the options array to make it more natural?
> 
> According to Vue documentation:
> 
> ![image](https://user-images.githubusercontent.com/36978416/128303486-a89089a5-8c55-4bc2-b0d2-c1217ee8b00a.png)

Noted, I was talking about should we do that to make it more flexible?

**Comment by SimonaliaChen** [comment]
We can separate `config` and `options` watcher to fix it. 
```ts
watch(
      [ config, () => props.options ],
      ([newConfig, newOptions], [oldConfig, oldOptions]) => {
        if (isEqualWithFunction(newConfig, oldConfig) && isEqual(newOptions, oldOptions)) return

        initStore()
      },
      { deep: true, immediate: true },
    )
```

to

```ts
watch(
     config,
      (newConfig, oldConfig) => {
        if (isEqualWithFunction(newConfig, oldConfig)) return

        initStore()
      },
      { deep: true, immediate: true },
    )

watch(
    () => props.options,
    initStore,
     { deep: true, immediate: true },
)
```

But there are still some potential problems such as  when mutating `config` without replacing its reference same problem remains. I wonder if previous fix #2344  made sense.
